[kotlin][client] fix encoding of individual parts of a multipart request

koscejev opened the pull request:
Fixes #11910

This propagates param.contentType field from individual API endpoint definitions to the actual ApiClient and uses it to create RequestBody for each individual part with correct Content-Type header as well as correct serialization of the individual part's body. If not specified, the default Content-Type is "application/octet-stream" for files and "application/json" for models - see https://swagger.io/docs/specification/describing-request-body/multipart-requests/. The actual serialization of individual part's body is now reusing code for serialization of the non-multipart body, so it now properly handles many cases.

^ This part was also verified on our own internal project using ok-http4 library only.

Potential questions/risks:
- I'm not 100% sure about how this works for part bodies that are neither files, nor models. But since it reuses code that is handling non-multipart body, so it is consistent with that at least.
- Some samples still contain old code, even though other samples were updated by the specified script? How does this work?
- Should this go to `master` instead? I will rebase as necessary.


### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jimschubert, @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m

koscejev commented:
Can someone explain what's failing in that build? It seems to fail for completely unrelated stuff, I don't fully understand it.

```
go: downloading github.com/gorilla/mux v1.7.3
?   	github.com/GIT_USER_ID/GIT_REPO_ID	[no test files]
# golang.org/x/net/http2
/home/circleci/.go_workspace/src/golang.org/x/net/http2/transport.go:417:45: undefined: os.ErrDeadlineExceeded
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (go-get) on project GoGinServer: Command execution failed. Process exited with an error: 2 (Exit value: 2) -> [Help 1]
```

koscejev commented:
@4brunu thanks for checking this out! In the mean time I found out that array/list of files is also not handled properly as it should be individual parts with the same name in the multipart request body, not just the whole list dumped as a single part. I will handle both problems, debug on our project, and then come back. Hopefully tomorrow.

4brunu commented:
Sure, take your time 🙂

koscejev commented:
I see all the code checks pass, but there are still some problems with samples/docs for other generator libraries. Unfortunately, I had some problem generating those, since I'm on Windows and even with Git Bash some stuff is generated incorrectly (e.g., with backslashes in paths), so I didn't push those unrelated changes.

koscejev commented:
I had to run `./bin/meta-codegen.sh` to update a file that made some check fail. Perhaps this could be added to PR checklist?

wing328 commented:
I've changed the target branch to the current master, which will be released as v6.0.0.

Can you please resolve the merge conflicts when you've time?

(v5.4.0 was released like a month ago)

koscejev commented:
@wing328 thanks for checking and fixing this, I was a bit clueless about whether it perhaps should go into 5.4.1 or something. I've rebased the changes. I also found and fixed one more small bug - but I can submit it in a separate PR, if necessary.
